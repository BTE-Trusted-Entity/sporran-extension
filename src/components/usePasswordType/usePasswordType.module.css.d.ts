declare namespace UsePasswordTypeModuleCssNamespace {
  export interface IUsePasswordTypeModuleCss {
    hide: string;
    show: string;
    toggle: string;
  }
}

declare const UsePasswordTypeModuleCssModule: UsePasswordTypeModuleCssNamespace.IUsePasswordTypeModuleCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: UsePasswordTypeModuleCssNamespace.IUsePasswordTypeModuleCss;
};

export = UsePasswordTypeModuleCssModule;
