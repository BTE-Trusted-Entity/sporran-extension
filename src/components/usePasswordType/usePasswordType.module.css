.toggle {
  composes: buttonIcon from '../../components/Button/Button.module.css';
  width: 2.25rem;
  height: 2.25rem;
  background-size: 1.25rem;
  position: absolute;
  right: 0.25rem;
}

.show {
  composes: toggle;
  background-image: url('../images/eye-closed.svg');
}

.hide {
  composes: toggle;
  background-image: url('../images/eye-opened.svg');
}
