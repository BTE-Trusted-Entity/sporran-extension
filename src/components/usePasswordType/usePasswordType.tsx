import { browser } from 'webextension-polyfill-ts';

import * as styles from './usePasswordType.module.css';

import { useBooleanState } from '../../utilities/useBooleanState/useBooleanState';

interface usePasswordTypeType {
  passwordType: 'password' | 'text';
  passwordToggle: JSX.Element;
}

export function usePasswordType(initiallyVisible = false): usePasswordTypeType {
  const t = browser.i18n.getMessage;
  const visible = useBooleanState(initiallyVisible);

  const passwordType = visible.current ? 'text' : 'password';

  const passwordToggle = visible.current ? (
    <button
      type="button"
      onClick={visible.off}
      className={styles.hide}
      title={t('component_usePasswordType_hide')}
      aria-label={t('component_usePasswordType_hide')}
    />
  ) : (
    <button
      type="button"
      onClick={visible.on}
      className={styles.show}
      title={t('component_usePasswordType_show')}
      aria-label={t('component_usePasswordType_show')}
    />
  );

  return { passwordType, passwordToggle };
}
