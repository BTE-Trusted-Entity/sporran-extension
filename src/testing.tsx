export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
