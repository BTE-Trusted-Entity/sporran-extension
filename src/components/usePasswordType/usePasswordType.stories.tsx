import { Meta } from '@storybook/react';

import { usePasswordType } from './usePasswordType';

export default {
  title: 'components/usePasswordType',
} as Meta;

export function Template(): JSX.Element {
  const { passwordType, passwordToggle } = usePasswordType();
  return (
    <form>
      <input type={passwordType} />
      {passwordToggle}
    </form>
  );
}
