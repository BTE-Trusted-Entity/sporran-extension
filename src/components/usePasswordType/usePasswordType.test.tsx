import userEvent from '@testing-library/user-event';

import { render, screen } from '../../testing/testing';

import { usePasswordType } from './usePasswordType';

function TestComponent() {
  const { passwordType, passwordToggle } = usePasswordType();
  return (
    <form>
      <input type={passwordType} />
      {passwordToggle}
    </form>
  );
}

describe('usePasswordType', () => {
  it('should provide values for rendering and toggle them', async () => {
    const { container } = render(<TestComponent />);
    expect(container).toMatchSnapshot();

    await userEvent.click(await screen.findByRole('button'));
    expect(container).toMatchSnapshot();

    await userEvent.click(await screen.findByRole('button'));
    expect(container).toMatchSnapshot();
  });
});
