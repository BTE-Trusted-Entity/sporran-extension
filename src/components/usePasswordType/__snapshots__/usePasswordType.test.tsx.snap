// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePasswordType should provide values for rendering and toggle them 1`] = `
<div>
  <form>
    <input
      type="password"
    />
    <button
      aria-label="Show"
      class="show"
      title="Show"
      type="button"
    />
  </form>
</div>
`;

exports[`usePasswordType should provide values for rendering and toggle them 2`] = `
<div>
  <form>
    <input
      type="text"
    />
    <button
      aria-label="Hide"
      class="hide"
      title="Hide"
      type="button"
    />
  </form>
</div>
`;

exports[`usePasswordType should provide values for rendering and toggle them 3`] = `
<div>
  <form>
    <input
      type="password"
    />
    <button
      aria-label="Show"
      class="show"
      title="Show"
      type="button"
    />
  </form>
</div>
`;
